DESCRIPTION

```diff
@@ -2,33 +2,34 @@ Package: leidenAlg
 Type: Package
 Title: Implements the Leiden Algorithm via an R Interface
 Version: 1.1.2
-Authors@R: c(person("Peter", "Kharchenko", email = "Peter_Kharchenko@hms.harvard.edu", role = c("aut")), 
-             person("Viktor", "Petukhov", email = "viktor.s.petukhov@ya.ru", role = c("aut")), 
+Authors@R: c(person("Peter", "Kharchenko", email = "Peter_Kharchenko@hms.harvard.edu", role = c("aut")),
+             person("Viktor", "Petukhov", email = "viktor.s.petukhov@ya.ru", role = c("aut")),
              person("Yichen", "Wang", email = "wayichen@umich.edu", role=c("aut")),
-             person("V.A.", "Traag", email = "v.a.traag@cwts.leidenuniv.nl", role = c("ctb")), 
-             person("Gábor", "Csárdi", email = "sardi.gabor@gmail.com", role = c("ctb")), 
-             person("Tamás", "Nepusz", email = "ntamas@gmail.com", role = c("ctb")), 
-             person("Minh Van", "Nguyen", email = "nguyenminh2@gmail.com", role = c("ctb")), 
+             person("Andrew", "Robbins", email = "robbiand@umich.edu", role=c("aut"))
+             person("V.A.", "Traag", email = "v.a.traag@cwts.leidenuniv.nl", role = c("ctb")),
+             person("Gábor", "Csárdi", email = "sardi.gabor@gmail.com", role = c("ctb")),
+             person("Tamás", "Nepusz", email = "ntamas@gmail.com", role = c("ctb")),
+             person("Minh Van", "Nguyen", email = "nguyenminh2@gmail.com", role = c("ctb")),
              person("Evan", "Biederstedt", email = "evan.biederstedt@gmail.com", role=c("cre", "aut")))
 Description: An R interface to the Leiden algorithm, an iterative community detection algorithm on networks. The algorithm is designed to converge to a partition in which all subsets of all communities are locally optimally assigned, yielding communities guaranteed to be connected. The implementation proves to be fast, scales well, and can be run on graphs of millions of nodes (as long as they can fit in memory). The original implementation was constructed as a python interface "leidenalg" found here: <https://github.com/vtraag/leidenalg>. The algorithm was originally described in Traag, V.A., Waltman, L. & van Eck, N.J. "From Louvain to Leiden: guaranteeing well-connected communities". Sci Rep 9, 5233 (2019) <doi:10.1038/s41598-019-41695-z>.
-License: GPL-3 
+License: GPL-3
 Copyright: See the file COPYRIGHTS for various leidenAlg copyright details
 Encoding: UTF-8
 LazyData: true
 Depends: R (>= 3.5.0), Matrix
-Imports: 
+Imports:
     graphics,
     grDevices,
     igraph,
     methods,
-    parallel, 
-    Rcpp (>= 1.0.5), 
+    parallel,
+    Rcpp (>= 1.0.5),
     sccore,
     stats
 Suggests:
     pbapply,
     testthat (>= 3.1.0)
-LinkingTo: Rcpp, RcppArmadillo, RcppEigen
+LinkingTo: Rcpp, RcppArmadillo, RcppEigen, igraph
```

Comment:
This we won't be able to do, I'm afraid

`L`$(R_HOME)/bin/Rscript -e "cat(find.package('igraph'))"`/libs/` won't work either

I've spent so many hours on igraph fixes over the years, as they make backwards incompatible changes.  You'll have to check the CHANGELOG. @theAeon

Reply:
not entirely sure what you mean-could you clarify?

Reply:
@theAeon You'll have to rummage through the GitHub issues and the versions + PRs to fully understand the issues. I'm a bit too traumatized to do it myself....

**I'll give you the takeaway, and then an explanation:**

**Takeaway is:** this R package 'leidenAlg' must remain independent of the R package 'igraph' and the C++ library (igraph) which that package uses. It cannot do any linking to that, at all. We explicitly wrote this code such that an old frozen version of igraph (C++) would be directly compiled and linked to. 

That's all we can do. This R package 'leidenAlg' and the C++ code within has successfully installed on our local versions of Linux, CRAN Mac and CRAN windows. 

My understanding is that we need to remove linking to `/urs/lib` directly due to 32-bit issues. That's reasonable. I'll do that on all R packages I have with C++ on CRAN/Bioconductor, and push others here to do the same (e.g. with numbat). 

This PR fixes your problem, but it gives me years of more problems. So, we won't link to igraph and/or the libraries within R igraph package. 

**Explanation:**
The igraph C++ library is constantly changing. And therefore the R igraph package is constantly changes. These changes are not backwards compatible. That means we would need to constantly update and change this R package 'leidenAlg'; that's a massive pain in the neck, and we don't have the time. It's a fragile hack to do `-L`$(R_HOME)/bin/Rscript -e "cat(find.package('igraph'))"``. If you snoop around the git history in these R packages, you'll see the problems.

Reply:
Okay-so I did a little bit of research re: igraph history and their future plans and I _totally_ see where you're coming from. Now definitely isn't the time to do this if you don't want the maintenance burden (though, i may maintain a friendly fork in the meantime)

the upshot is that apparently the igraph team has realized that "constantly breaking things" isn't a great approach to development and apparently a stable core has been on the roadmap for a few years per their [forum](https://igraph.discourse.group/t/igraph-roadmap-2021-2023/972). Probably worth coming back to at that point.

Reply:
@theAeon My understanding of [the R manual](https://cran.r-project.org/doc/manuals/R-exts.html#Package-Dependencies) is that `LinkingTo` is primarily for using _headers_, and there is no supported mechanism that allows one R package to link to a share library provided by another R package (the [Linking to native routines](https://cran.r-project.org/doc/manuals/R-exts.html#Linking-to-native-routines-in-other-packages) section describes a very specific and restricted way that's not suitable for what you're trying to do here).

The issue here is not with igraph, but that R (and CRAN) simply do not support the sort of packaging you are trying to do.

Regarding "igraph stability", and the work towards a stable C core: this work won't solve the problems here. The igraph C library already provides a stable API and ABI within a single release series (so within 0.10.x, within 0.9.x, etc.) "Stable ABI" here means that a program will work with any version of the igraph shared library (within a series, say 0.10.x) without the need to recompile. For example, Debian don't need to recompile all of igraph's dependencies when upgrading from 0.10.8 to 0.10.9. However, it does _not_ mean that you can load two different versions of the igraph shared library, create an in-memory object with one, and manipulate it with the other.

Thus, for packages like leidenAlg, there are two options:

 - Ship their own copy of igraph and make a full copy of the internal graph object anytime they manipulate it. This is the current approach. Yes, this does bring some overhead.
 - Link to the same igraph C library as the igraph R package does. This is not possible at the moment. It may become possible eventually, but won't solve the problem here, as R's packaging system (and CRAN) don't support shipping a shared library that multiple R packages can link to. We can't do anything on the igraph side to change this.

At the risk of sounding like a broken record, I'll say it once more:

 - Look at what you need leidenAlg for in practice
 - Check how many of the use cases you can replace with igraph's built-in Leiden function. You will find that this is possible for the great majority of practical biology applications.

Reply:
>Thus any solution similar to what you're doing here will be a fragile hack that is susceptible to being broken by changes R (and CRAN) make.

That's correct, I agree with this

The better solution for me would be to figure out why this doesn't compile properly on your flavor of Unix; following up with email